sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -130,6 +130,14 @@ case class Join(
   }
 }
 
+/**
+ * A hint for the optimizer that we should broadcast the `child` if used in a join operator.
+ */
+case class BroadcastHint(child: LogicalPlan) extends UnaryNode {
```

Comment:
+1
Add specific node is also doable, probably the node can be named like `Hint` in `LogicalPlan`, and could be more than `broadcast`, like `uniq_key` etc. 

Any idea?